shopping-cart/app/com/commercetools/sunrise/framework/checkout/payment/DefaultCheckoutPaymentControllerAction.java

```diff
@@ -16,21 +17,23 @@
 import io.sphere.sdk.payments.PaymentMethodInfo;
 import io.sphere.sdk.payments.commands.PaymentCreateCommand;
 import io.sphere.sdk.payments.commands.PaymentDeleteCommand;
-import io.sphere.sdk.payments.queries.PaymentByIdGet;
-import io.sphere.sdk.utils.CompletableFutureUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import play.libs.concurrent.HttpExecution;
 
 import javax.inject.Inject;
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.concurrent.CompletionStage;
 
-import static java.util.Collections.emptyList;
 import static java.util.stream.Collectors.toList;
 
+/**
+ * Deletes previous payments associated with the cart and adds the new one.
+ * {@code paymentInfo.payments} must be expanded in order to properly work, otherwise previous payments will not be removed.
```

Comment:
Ah actually it was not a class, but the expansion path as it is expected by the platform :)
I will rewrite it to make it clear that I'm talking about paths.